gpio/__init__.py

```diff
@@ -0,0 +1,304 @@
+# -*- coding: utf-8 -*-
+__version__ = '1.0.0'
+
+from threading import Lock
+import os
+
+
+_export_lock = Lock()
+_open_pins = {}
+
+
+GPIO_ROOT = '/sys/class/gpio'
+GPIO_EXPORT = os.path.join(GPIO_ROOT, 'export')
+GPIO_UNEXPORT = os.path.join(GPIO_ROOT, 'unexport')
+FMODE = 'w+'  # w+ overwrites and truncates existing files
+IN, OUT = 'in', 'out'
+LOW, HIGH = 0, 1
+
+
+class GPIOPin(object):
+    """Handle pin state.
+
+    Keeps track of file nodes and functions related to a pin.
+
+    Args:
+        pin (int): the GPIO pin to configure
+    """
+    def __init__(self, pin, mode=None, initial=LOW, active_low=None):
+        try:
+            # Implicitly convert str to int, ie: "1" -> 1
+            pin = int(pin)
+        except TypeError:
+            raise TypeError("pin must be an int")
+
+        if _open_pins.get(pin):
+            raise RuntimeError("pin {} already configured".format(pin))

```

Comment:
add "Use configured(pin) instead"

Reply:
Rewritten most of this, I think. I'm losing the plot a little. `setup` is still awful. You might have got me on that one :laughing:

Reply:
`setup` has fallen. Long live `setup`.